pkg/signup/tokenparser_test.go

```diff
@@ -0,0 +1,68 @@
+package signup_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/signup"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTokenParser(t *testing.T) {
+	// create logger and registry.
+	logger := log.New(os.Stderr, "", 0)
+	configRegistry := configuration.CreateEmptyRegistry()
+
+	// test key
+	keyJSON := `{"keys":[
+		{"kid":"key0","kty":"RSA","e":"AQAB","n":"nzyis1ZjfNB0bBgKFMSvvkTtwlvBsaJq7S5wA-kzeVOVpVWwkWdVha4s38XM_pa_yr47av7-z3VTmvDRyAHcaT92whREFpLv9cj5lTeJSibyr_Mrm_YtjCZVWgaOYIhwrXwKLqPr_11inWsAkfIytvHWTxZYEcXLgAXFuUuaS3uF9gEiNQwzGTU1v0FqkqTBr4B8nW3HCN47XUu0t8Y0e-lf4s4OxQawWD79J9_5d3Ry0vbV3Am1FtGJiJvOwRsIfVChDpYStTcHTCMqtvWbV6L11BWkpzGXSW4Hv43qa-GSYOD2QU68Mb59oSk2OB-BtOLpJofmbGEGgvmwyCI9Mw"},
+		{"kid":"key1","kty":"RSA","e":"AQAB","n":"4niTFsMZ_gLOcg9OuwMK4LpBzpdS8ulIGmx5B4rNqWVHAWMpg4kEmZTQffVmKmiw3NUDSaSWLcLJp22ekN2sj1E7tEu1pJksYsXNDa3WLaE1uqVeso-HVv2rIbucd5xMaryvf490g2I-PSrZdSvN73VqJM525s7pPanxe1skqh8"}
+	]}`
+	// test JWT, signed with above keys
+	jwt0 := "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImtleTAifQ.eyJqdGkiOiIwMjgyYjI5Yy01MTczLTQyZDgtODE0NS1iNDVmYTFlMzUzOGIiLCJleHAiOjAsIm5iZiI6MCwiaWF0IjoxNTE3MDE1OTUyLCJpc3MiOiJ0ZXN0IiwiYXVkIjoiY29kZXJlYWR5LXJlZ2lzdHJhdGlvbi1zZXJ2aWNlIiwic3ViIjoiMjM5ODQzOTgtODU1YS00MmQ2LWE3ZmUtOTM2YmI0ZTkyYTBjIiwidHlwIjoiQmVhcmVyIiwic2Vzc2lvbl9zdGF0ZSI6ImVhZGMwNjZjLTEyMzQtNGE1Ni05ZjM1LWNlNzA3YjU3YTRlOSIsImFjciI6IjAiLCJhbGxvd2VkLW9yaWdpbnMiOlsiKiJdLCJhcHByb3ZlZCI6dHJ1ZSwiZW1haWxfdmVyaWZpZWQiOnRydWUsIm5hbWUiOiJUZXN0MCBVc2VyMCIsImNvbXBhbnkiOiJUZXN0IENvbXBhbnkgMCIsInByZWZlcnJlZF91c2VybmFtZSI6InRlc3R1c2VyMCIsImdpdmVuX25hbWUiOiJUZXN0MCIsImZhbWlseV9uYW1lIjoiVXNlcjAiLCJlbWFpbCI6InRlc3R1c2VyMEB0ZXN0LnQifQ.a2JbOJXEB7IAK4JUEcW886VNJoQeHJ_yQiN5dgoPECpKw7PdQAajVBEnSRj6irDsNCjBznDnthSg_rfRv10_cN5K--FoB3eynylMm2Zwdj1JiVpiYd-lK0OfdLUfXlWalq-GLAoiI0UzDBjQE_Izzr58KaeYzjP7Gb9et99zVLDpnmLCoa9vCrXL7G-Dir4l2CqVtorp653Y19EY-A4TeJEitrqg4Uc8SXGskgpXnDZogX0IzFnwvDXawxeUUUoVaSugaxeyAh-ZyFMOFMtHXOM5M-7_3_kMMeVlKrCrbMRN6as5-jQ2vyrZND6fnVuZhrGgoIZNYfN5qaJHgpMikQ"
```

Comment:
you could probably use constants here and share them with between the `tokenparser_test.go` and `keymanager_test.go`

Reply:
I intentionally did not do that as there may be more and/or different tokens coming in in later PRs.

Reply:
ok

Reply:
While I think it's OK to hardcode some tokens in our tests for testing invalid token signatures it would be pretty hard to maintain the tests if we use hardcoded tokens which are supposed to have valid signatures.
Instead we should generate tokens in our tests programmaticaly. Take a look at https://github.com/fabric8-services/fabric8-common/blob/master/test/auth/auth.go for example. There are a bunch of utils which helps to generate tokens for tests using pre-defined test keys. So, in your tests you can always focus on the token content. 

And if you need to change some token claim in the test you don't need to manually re-encode the token. It also helps a lot with code readability. And if you need to change the test key you don't have to re-encode all the tokens, etc.

Reply:
Done 5f83a27. If the goal there was to make things easier and light, we failed :-)

Reply:
Looks better now. Thanks.

> If the goal there was to make things easier and light, we failed 

I will ask you again if it's easier to use when you will be adding new tests for future code or modifying the existing ones or debugging them ;)
No any manual token generation and parsing needed anymore.